.github/copilot-instructions.md

```diff
@@ -0,0 +1,105 @@
+    - Metadata `ai-usage: ai-assisted` if any AI assistance was used
+    - Metadata `ms.date: <today's date>` with a format of DD-MM-YYYY.  If the file already has a `ms.date` metadata, update it to today's date if more than 50 characters have changed in the file.
+    
+## Version Targeting
+
+### Detection Priority Order
+1. **Explicit PR/Issue Instructions** - Prioritize explicit version ranges mentioned
+2. **File Metadata** - Check YAML frontmatter for specifications
+3. **Inline Moniker Tags** - Look for version range specifications
+4. **Repository Branch Context** - Consider branch where PR will be merged
+5. **Directory Structure** - Check for version-specific directories
+
+### Common Range Patterns
+- Fixed Range: `>= aspnetcore-7.0 <= aspnetcore-9.0`
+- Open Upper Bound: `>= aspnetcore-7.0`
+- Open Lower Bound: `<= aspnetcore-9.0`
+- Specific Version: `== aspnetcore-9.0`
+
+## API References and Verification
+
+- Never guess API documentation IDs, always verify.  If unable to verfy, say so.
+- Check if features were introduced after .NET Core 3.1
+- Default to most compatible syntax when uncertain
+- For breaking changes, check:
+  - Changes to default behavior of existing APIs
+  - Removed or renamed public API elements
+  - Modified method signatures or return types
+  - Changed serialization formats
+  - Altered dependency requirements
+  - Changes to configuration schema
+
+## Links and References
+
+- Use **relative links** for files within this repo
+- For learn.microsoft.com links, **remove** `https://learn.microsoft.com/en-us` from URLs
+- Use cross-references for APIs: `<xref:api-doc-ID>`
+- Get doc ID from the relevant XML doc in dotnet-api-docs
+- Omit first two characters of the doc ID
+
+## Markdown File Naming and Organization
+
+- If you're adding a new Markdown file, it should be named in all lowercase with hyphens separating words. Also, omit any filler words such as "the" or "a" from the file name.
+  
+## Code Snippets
+
+- For snippets **longer than 6 lines**:
+  - Place in separate `.cs` file in a `snippets` folder next to the Markdown file
+  - Create a subfolder named after the document
+  - For version-specific code, include a version folder
+  - Add a simple `.csproj` file targeting the appropriate .NET version
+
+- Reference snippets using triple-colon syntax:
+  - **Use file-relative paths** for snippets specific to a single doc file:
+    ```
+    :::code language="csharp" source="../snippets/my-doc/Program.cs":::
+    ```
+  - **Use repository root-relative paths** for shared snippets:
+    ```
+    :::code language="csharp" source="~/tutorials/min-web-api/samples/9.x/todoGroup/TodoDb.cs":::
+    ```
+- For longer snippets, highlight specific lines:
+  ```
+  :::code language="csharp" source="~/path/to/file.cs" range="5-10" highlight="2-3":::
+  ```
+- Use modern C# coding patterns in all examples
```

Comment:
Possibly, the following elaboration for Line 85? .... plus a table of code languages and indentation? .......

```suggestion
- Use the latest, non-preview C# coding patterns in non-preview code examples
- Use the latest preview C# coding patterns in preview code examples

The following table indicates the code language to use for markdown code blocks or the `language` attribute of a code snippet and the amount of indentation to use.

Content of the snippet | Code language | Indentation in spaces
:---: | :---: | :---:
C# | csharp | 4
HTML | html | 4
CSS | css | 4
JavaScript | javascript | 2 (4 spaces for line continuation)
XML | xml | 2
JSON | json | 2
Console | console | 2
Text | - | 2
```